Source/WebCore/platform/graphics/FontPlatformData.h

```diff
@@ -126,7 +126,11 @@ class FontPlatformData {
 
     static FontPlatformData cloneWithOrientation(const FontPlatformData&, FontOrientation);
     static FontPlatformData cloneWithSyntheticOblique(const FontPlatformData&, bool);
+
+    // cloneWitSize to be implemented by each platform since it needs to re-instantiate the platform font object.
     static FontPlatformData cloneWithSize(const FontPlatformData&, float);
+    static FontPlatformData cloneWithAdjustedSize(const FontPlatformData&, float);
```

Comment:
Unfortunate that this has to copy the source when the size is already correct. Consider renaming and changing the argument to `FontPlatformData&&`? I understand this fits in with the existing clone functions, but most of those seem to copy unconditionally, so this is different.

One way to do this is to change this into a member function that modifies a `FontPlatformData` in place.

Reply:
I don't think this function needs to be public.

Reply:
Good point. However, I've just realized `FontPlatformData` doesn't have a move constructor, which also makes the move in `fontPlatformDataWithAdjustedSize`. Should we have a move ctor for it or there's a reason for not having one? @litherum

Reply:
I've implemented the in-place version and modified cloneWithSize functions to use it.